### ISSUE TITLE
Style/EmptyElse, Style/IfUnlessModifier: Correcting conditional assignment with nil leads to bug where assignment doesn't occur

### ISSUE THREAD

**Issue by jnebeker** [rubocop-hq/rubocop#6376]
Correcting `Style/EmptyElse` and `Style/IfUnlessModifier` violations in a conditional assignment leads to unintended bug where the assignment doesn't occur. 

--------

## Expected behavior

With the default RuboCop configuration the following conditional assignment shouldn't trigger an `Style/EmptyElse` violation.

```ruby
self.ivar = if condition?
              "ivar"
            else
              nil
            end
```

## Actual behavior

The code triggers a `Style/EmptyElse` and subsequent `Style/IfUnlessModifier` which lead to unintended behavior when corrected.

## Steps to reproduce the problem

This is a two-step bug that occurs for the following snippet of code where we want `self.ivar` to be set to `nil` if the condition is false:

```ruby
self.ivar = if condition?
              "ivar"
            else
              nil
            end
```

First, this triggers a valid `Style/EmptyElse` violation.  Refactoring the code to resolve that looks like this:

```ruby
self.ivar = if condition?
              "ivar"
            end
```

This code triggers a `Style/IfUnlessModifier` violation which can be resolved by refactoring the code like this:

```ruby
self.ivar = "ivar" if condition?
```

Except now the behavior has changed, in this form the assignment won't happen at all unless the condition is true.  The original intention was to assign `self.ivar` to `nil` if the condition was false.  This isn't an issue for local variables that haven't been set before because they will still be `nil` by default, but in this case there could be an initial value for the ivar which is still set after this conditional assignment runs, instead of it being set to `nil` as we originally intended.

It's true that this problem can be resolved by the use of parentheses or with a ternary, but the original form that is explicitly setting the variable to `nil` if the condition isn't met is clearer and easier to understand. 

## RuboCop version
```
$ bundle exec rubocop -V
0.59.2 (using Parser 2.5.1.2, running on ruby 2.2.3 x86_64-darwin14)
```
